src/typing/env_js.ml

```diff
@@ -29,6 +29,10 @@ let refinement_key names =
 let is_refinement name =
   (String.length name) >= 5 && (String.sub name 0 5) = "$REFI"
 
+let refined_object_from_key name =
+  let i = String.index name '.' in
+  String.sub name 6 (i - 6)
```

Comment:
Definitely need some length checks and so on here - Ocaml is pretty throw-happy on out of range errors and such.

Per another discussion we had recently, I'd move all the key-wrangling stuff into Type_inference_js.Refinement. (Note that I haven't gotten to where this is actually used, may have more useful stuff to say then.)